zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/bpmn/container/MultiInstanceBodyProcessor.java

```diff
@@ -213,19 +214,19 @@ public void afterExecutionPathCompleted(
       }
       return;
     }
+    final var inputCollectionOrFailure = readInputCollectionVariable(element, flowScopeContext);
+    if (inputCollectionOrFailure.isLeft()) {
+      // this incident is un-resolvable
+      incidentBehavior.createIncident(inputCollectionOrFailure.getLeft(), childContext);
+      return;
```

Comment:
❌ As this incident is unresolveable, we must protect against it earlier. This requires checking the input collection in the `beforeExecutionPathCompleted` method also for the non-sequential (parallel) case.

Reply:
I introduced no change here, this has been here from before, and I'm not comfortable changing this behavior in the scope of this PR.

Reply:
And just as an FYI: we do create an incident  at `onChildActivating`

Reply:
if what you mean is to remove the lines 
https://github.com/camunda/camunda/blob/008d8b9106088d19d4895936b6846254d019b1c9/zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/bpmn/container/MultiInstanceBodyProcessor.java#L185-L187

I shall do so, but I'm not sure if it's necessary